mne/preprocessing/maxwell.py

```diff
@@ -0,0 +1,526 @@
+
+    if nmeg <= 0:
+        raise RuntimeError('Could not find any MEG channels')
+
+    meg_info = pick_info(info, picks) if nmeg > 0 else None
+
+    # Create coil descriptions with transformation to head or MRI frame
+    if elekta_defs:
+        elekta_coil_defs = op.join(op.split(__file__)[0], '..', 'data',
+                                   'coil_def_Elekta.dat')
+        templates = _read_coil_defs(elekta_coil_defs)
+
+        # Check that we have all coils needed
+        template_set = set([coil['coil_type'] for coil in templates['coils']])
+        req_coil_set = set([coil['coil_type'] for coil in meg_info['chs']])
+        if not req_coil_set.issubset(template_set):
+            logger.info('Didn\'t locate find enough Elekta coil definitions,'
+                        ' using default MNE coils.')
+            templates = _read_coil_defs()
+    else:
+        templates = _read_coil_defs()
+
+    if nmeg > 0:
+        # TODO: In fwd solution code, reformulate check that forces head
+        # coords and remove this hack. (Or use only head coords)
+        # Uncomment below for device coords
+        # info['dev_head_t']['trans'] = np.eye(4)
+        megcoils = _create_coils(megchs, accuracy, info['dev_head_t'], 'meg',
+                                 templates)
+
+    return megcoils, meg_info
+
+
+def _update_info(raw, origin, int_order, ext_order, nsens, nmoments):
+    """Helper function to update info after Maxwell filtering.
+
+    Parameters
+    ----------
+    raw : instance of mne.io.Raw
+        Data to be filtered
+    origin : array-like, shape (3,)
+        Origin of internal and external multipolar moment space in head coords
+        and in millimeters
+    int_order : int
+        Order of internal component of spherical expansion
+    ext_order : int
+        Order of external component of spherical expansion
+    nsens : int
+        Number of sensors
+    nmoments : int
+        Number of multipolar moments
+
+    Returns
+    -------
+    raw : mne.io.Raw
+        raw file object with raw.info modified
+    """
+    from .. import __version__
+    # TODO: Flesh out/fix bookkeeping info
+
+    raw.info['maxshield'] = False
+    sss_info_dict = dict(in_order=int_order, out_order=ext_order,
+                         origin=origin, nsens=nsens, nmoments=nmoments,
+                         components=np.ones(nmoments))
+
+    max_info_dict = dict(max_st={}, sss_cal={}, sss_ctc={},
+                         sss_info=sss_info_dict)
+
+    block_id = dict(machid=-1 * np.ones(2), secs=-1, usecs=-1, version=-1)
+    proc_block = dict(max_info=max_info_dict, block_id=block_id,
+                      creator='mne-python v%s' % __version__,
+                      date=-1, experimentor='')
+
+    # Insert information in raw.info['proc_info']
+    if 'proc_history' in raw.info.keys():
+        raw.info['proc_history'].insert(0, proc_block)
+    else:
+        raw.info['proc_history'] = [proc_block]
```

Comment:
Note. @wronk @Eric89GXL we should check that MNE-Python SSSing IO works nicely with our covariance code.